### DISCUSSION TITLE
How can i move data from an  prop to anther?

### DISCUSSION THREAD

**Discussion by bestony** [alpinejs/alpine#1092] [1. Help] [answered]
## WHAT i WANT TO DO?

I have two difference array in my Alpine.js Instance, and i want to move one element to another.

## WHAT i try?

I try to use foreach to find element, and use push push to another array ,but my data will be wrong

![image](https://user-images.githubusercontent.com/13283837/107836738-2752ca00-6dd9-11eb-9e97-f7e9c27f1505.png)

it should be two object, but be one plain object with one Proxy Object

## WHAT i want?

1. some code or some text for my status
2. some example
3. other can give some help.

## Some code of mine

```
function dataFunc(){
 return {
   hosts:[],
   members:[],
   async _moveUser(cmd) {
    let newHost;
    let oldHosts = this.hosts;
    this.members.forEach(member => {
        if (member.id == cmd.id) {
            newHost = member;
        }
    });
    oldHosts.push(newHost)
    this.hosts = oldHosts;
    console.log("ok", this.hosts);
  }
}
```

**Comment by KevinBatdorf** [comment, marked as answer]
`this.hosts` will be a proxy because Alpine is observing it for changes. 

You can treat it the same as a normal object though. Are you having an issue or just surprised while checking console.log?

**Reply by bestony** [reply]
yes, i move it ,and i get a array mix object and proxy object.

like this image, it not work good like i put two object in array.

![image](https://user-images.githubusercontent.com/13283837/107838243-e7dbac00-6ddf-11eb-923f-620919e3c61d.png)

**Reply by KevinBatdorf** [reply]
I understand. When you do this: `let oldHosts = this.hosts;` you are not creating a new object, but are creating a reference to an existing object. Then, when you do this: `oldHosts.push(newHost)` you are pushing a second object. 

Change this: `let oldHosts = this.hosts;`
To this: `let oldHosts = JSON.parse(JSON.stringify(this.hosts));`

**Reply by bestony** [reply]
thx! It Works!

**Comment by SimoTod** [comment]
Is there a reason why you simply don't call this.hosts.push(member) inside the loop and get read of newHost and oldHost? That would be faster and use less memory because you don't need to rebuild a new array every time.
